### DISCUSSION TITLE
Sylph Faction

### DISCUSSION THREAD

**Discussion by evanbowman** [evanbowman/skyland-beta#126] [Ideas]
Discussion related to design of a playable Sylph faction

**Comment by evanbowman** [comment]
![slyph_cannon](https://github.com/user-attachments/assets/31bbdeaa-2860-424c-91a8-14e2f0851dcf)
Icon for some kind of Sylph cannon

**Comment by evanbowman** [comment]
I'm considering how to redesign the game's dependency tree when adding new factions. Currently, some weapons and other things that you can build require a workshop. If I add a sylph-workshop block, then what about the existing weapons that depend on a workshop? Will players be unable to build those when playing as the Sylph? Do I design a full set of new blocks? I'm considering ideas.

**Reply by evanbowman** [reply]
Maybe instead, I don't create separate workshop/manufactory etc. blocks for each faction, and certain blocks are labelled in the glossary as faction specific. Like blocks labeled human-only, sylph-only, etc.

**Reply by Doronimmo** [reply]
Maybe you could do both - have different workshops and factories, but keep a bunch of base blocks that could be built by any faction.

**Comment by Doronimmo** [comment]
Some ideas:
Special sylph hull resistant to explosions (the explosion will not expand to the other blocks).
Sylph could build a deflector (maybe overpowered?)
Some weapons can have an amplifier module attached (less cooldown)
Ability for crew to repair all blocks touching the castle (probably overpowered)
When doing the sylph quest, their language is normal.
Regular things shaped differently - could change the familiar castle builds and force player to adapt.

**Reply by evanbowman** [reply]
Interesting ideas!
I also thought of:
phase-shifter: when activated, sylph castle becomes translucent for a few seconds and dodges attacks
sylph-cannon: does damage equal to 25% of a block's max health

**Comment by evanbowman** [comment]
I've started working on some stuff over on this branch: <https://github.com/evanbowman/skyland-beta/tree/factions>

![ezgif-2-9707a440dc](https://github.com/user-attachments/assets/49240460-e69e-41fe-8616-79f5dac1b6ac)

**Reply by Doronimmo** [reply]
I see that the radar is human only, is the level generation going to respect that?

**Reply by evanbowman** [reply]
> I see that the radar is human only, is the level generation going to respect that?

Actually, now that I think about that choice some more, it would affect your ability to plunder castles effectively as goblins, so I think I'll undo that change.

**Comment by evanbowman** [comment]
While I originally didn't want to change level scripts too much when adding factions, I think I will end up making some more changes. E.g. when playing as goblins, there should be more opportunities to be evil, e.g. options for attacking neutral islands, raiding shops, conscripting crew instead of rescuing them, etc. I'm going to focus on gameplay changes first, but eventually I might change other things.

**Reply by Doronimmo** [reply]
Might be funny for a goblin event to encounter the human ship, and your crew will surrender.

**Reply by evanbowman** [reply]
That's a neat idea. Mainly I'm thinking of the fact that most existing events involve you recruiting human crew, so when playing as goblins/sylph, you'd still have a mostly human crew. I'm thinking of swapping these events out in particular based on faction.

**Comment by evanbowman** [comment]
Furthermore, I will need to make a clearer visual indicator of which faction you're playing as. Currently, it's difficult to tell. Maybe crew uniforms will be a different color, I'm not sure yet.

**Reply by Doronimmo** [reply]
Might be difficult, but the different factions could have the passports styled differently.

**Reply by evanbowman** [reply]
That's not too difficult to do, I think. The color palette for passports is partly shared with the adventure-log, so I could change the color scheme for both of them, with some other adjustments to graphics and text.

**Comment by evanbowman** [comment]
I am working on the sylph faction on this branch: https://github.com/evanbowman/skyland-beta/tree/sylph-faction

Overall premise: the sylph are a technologically advanced society, but unfortunately, the energy crystals that supply power to their civilization have mysteriously started to fracture and degrade, leaving the sylph unexpectedly vulnerable to goblin raids.

The faction-specific scripts for zone 1 are finished, and I'm currently working on the quests:
- [ ] quests/0.lisp
- [x] quests/1.lisp -> keep the banana quest for now, it isn't faction-specific in any way
- [x] quests/2.lisp -> The farm colony quest doesn't need revisions for now
- [ ] quests/3.lisp
- [ ] quests/4.lisp
- [ ] quests/5.lisp
- [x] quests/6.lisp -> I've replaced this quest, where you originally have to rescue a sylph child, with an alternate event where you return a goblin child to his home. The brainwashed goblin kid is terrified that the sylph will cook him in one of their notorious goblin-cooking pots.
- [x] quests/7.lisp -> The mining quest doesn't need significant changes. Maybe I'll change the characters and dialog around a bit...
- [ ] quests/8.lisp -> TODO: replace atomics quest with an event where the Sylph Conclave sends you out to retrieve a particle-lance
- [x] Custom save screen for sylph
- [x] events/neutral/0/3.lisp

Furthermore:
Some of the events in zone 3 give you fancy sylph technology. But when you play as Sylph, you can build this stuff normally. TODO: replace these events with different ones, maybe something that gives you goblin or human specific stuff:
- neutral/2/6_2.lisp
- neutral/2/6_3.lisp
- neutral/2/6_4.lisp

**Comment by evanbowman** [comment]
Sylph savegame screen idea
<img width="480" height="320" alt="Skyland-405" src="https://github.com/user-attachments/assets/b517b42a-aaed-42f4-b194-e8f720a1f41f" />
